### ISSUE TITLE
[1.14.4] Drop key on empty inventory slot drops item in hand

### ISSUE THREAD

**Issue by SylentButDedly** [MinecraftForge/MinecraftForge#5968]
**Minecraft Version:** 1.14.3

**Forge Version:** 28.0.13    Note: I've experienced this in Forge 1.14.x builds since 1.14.2

**Logs:** N/A

**Steps to Reproduce:**
 1. Have an ItemStack in your hand
 2. Press your drop key on an empty inventory slot in the inventory screen
 3. Item in your hand will get tossed out

**Description of issue:**
If you press your drop key on an empty slot of your inventory it will drop what's in your hand. Have tested and it doesn't happen in vanilla. It only happens in the inventory screen, doesn't happen in chests and such. 

Video: https://youtu.be/QVlrEvFUoL8

Let me know if you need anything else!

**Comment by pau101** [comment]
So looking into it shows #5367 is the culprit, as it is allowing the unhandled key event to pass along to the main game key handling.

**Comment by mezz** [comment]
I looked at this with Paul and he is right. 

We can't use the return value from `IGuiEventListener#keyPressed`, it means something different. When that function returns false, vanilla stores the key and has the game loop handle the key press, which I did not realize when writing the PR for 1.13. 

We need to fix the implementation, one way is to re-implement the boolean flag for "key handled" from 1.12 and use that instead.
